### ISSUE TITLE
Complex html structures not rendering properly

### ISSUE THREAD

**Issue by zkrige** [readium/readium-js-viewer#686]
Complex html structures not rendering properly when increasing font size in reflowable epubs

#### Expected Behaviour
Rendering should match iBooks - code line 10 in image below
![screen shot 2018-02-07 at 12 24 35](https://user-images.githubusercontent.com/968412/35911802-8075566a-0c03-11e8-8fb7-25baa121b56b.png)

#### Observed behaviour
Instead Readium breaks layout slightly `10` is truncated on left
<img width="686" alt="screen shot 2018-02-07 at 12 36 56" src="https://user-images.githubusercontent.com/968412/35911847-a41bd4d6-0c03-11e8-92ee-28737dcbee4a.png">

#### Steps to reproduce
1) open attached epub with readium
2) browse to item_GuidedTour
3) change font size to 190%
4) go to page 4

#### Test file(s)
https://drive.google.com/open?id=1HI8ZstBLja7YjD7wOpQ5Rjx_-YtOW9xw

### Product
* Native application (Readium SDK C++)
  * official "launcher"
  * iOS 11.2

**Comment by danielweck** [comment]
I just tried with https://readium.firebaseapp.com (which is `readium-js-viewer`, which itself is based on the same `readium-shared-js` code as native iOS / Android reader apps). Everything looks okay in both paginated and scroll view modes:
![screen1](https://user-images.githubusercontent.com/625201/36216861-ded98150-11a7-11e8-8440-0467cec73450.png)
![screen2](https://user-images.githubusercontent.com/625201/36216868-e2273532-11a7-11e8-83c4-42d24369800e.png)

**Comment by zkrige** [comment]
Yes, it works fine at 100% font size, but if you increase the font size it goes wonky

**Comment by danielweck** [comment]
I suspect this problem occurs because of how Readium applies the font-size change to DOM fragments:
https://github.com/readium/readium-shared-js/blob/develop/js/helpers.js#L256
Applying a percentage value at the top-level does not work in many cases (because of CSS cascade overrides), so the current method is to walk the DOM tree, record the original computed size, then apply the desired "scale" factor as a calculated, not relative value.
Although the line height is proportionally adjusted, I think that your example reveals a similar / related issue with content "gutters". I'm not sure how to address this though...

**Comment by zkrige** [comment]
We've built our own epub reader and managed to get around this by applying `webkit-text-size-adjust` to the `html` tag - it seems to work on the entire document

**Comment by danielweck** [comment]
Thanks for the tip! https://developer.mozilla.org/en-US/docs/Web/CSS/text-size-adjust
